### DISCUSSION TITLE
crop_segmentation model works for area other then US?

### DISCUSSION THREAD

**Discussion by mursaleen-bkk** [microsoft/farmvibes-ai#72] [Q&A] [answered]
Sir, I want to find the area covered by crops in pakistan. For That I want to use crop segmentation model. Will it work for pakistan?

**Comment by rafaspadilha** [comment, marked as answer]
Hi, @mursaleen-bkk. Thank you for using FarmVibes.AI.
 
The examples in `notebooks/crop_segmentation` show you how to train a model for crop land segmentation. As of now, we do not provide a pre-trained model. 

The notebook examples generate training data from NDVI stacks and training labels from CropDataLayer (CDL) maps. Unfortunately, CDL is only available to the Continental USA. In your scenario, you would need to use ground-truth labels available for Pakistan from external sources. If you have those, you can combine them with FarmVibes.AI workflows with global availability (e.g., Sentinel, Landsat) for generating input training data.

**Reply by Amr-MKamal** [reply]
Hello , @rafaspadilha , in the case of integrated external sources to act as CDL can you explain in more details how could this be done , also what about training the model on CDL maps & then use it to infer a new region CDL map & then use this map to train the system again ?

**Reply by rafaspadilha** [reply]
In the examples, FarmVibes.AI downloads and preprocesses the input images and labels that will guide the model training. In that case, CDL is used as the ground-truth data, while stacks of NDVI data act as the input images. 

In a scenario of an external source of labeled data, FarmVibes.AI workflows would only download the input data (NDVI as we do in our examples, or other data modalities, such as raw Landsat or Sentinel-2 imagery) and you/the external source would provide the labels to guide the training. This would most likely require adaptations to the code of: 

- `notebooks/crop_segmentation/notebook_lib/datasets.py`: to create a new `RasterDataset` for the ground-truth data and define any required pre-processing;
- `notebooks/crop_segmentation/notebook_lib/modules.py`: to setup the `DataModule` considering the new labelled data;

With respect to the type and potential sources of external data, a crop land segmentation model would required annotated data for the fields and lands containing crops, similar in nature to CDL.  The model trained in the notebook examples is optimized with imagery originated from a 5,000 km2 region in the US and we have not yet evaluated its generalization across other geographic regions and crop types.
